cmd/kusk/cmd/upgrade.go

```diff
@@ -89,91 +88,94 @@ var upgradeCmd = &cobra.Command{
 			switch deployment.Name {
 			case "kusk-gateway-manager":
 				if !utils.IsUptodate(getVersions(deployment.Name, "manager", deployment)) {
-					kuskui.PrintStart("kusk is already installed. Upgrading...")
+					kuskGatewaySpinner := utils.NewSpinner("Upgrading Kusk Gateway...")
 					for _, c := range deployment.Spec.Template.Spec.Containers {
 						if c.Name == "manager" {
 							if err := applyk(dir); err != nil {
-								kuskui.PrintError("❌ failed upgrading kusk", err.Error())
+								kuskGatewaySpinner.Fail("Failed upgrading Kusk Gateway", err.Error())
 								reportError(err)
 								return err
 							}
-							kuskui.PrintStart("kusk upgraded")
 						}
 					}
 
 					if err := utils.WaitForPodsReady(cmd.Context(), c, namespace, name, time.Duration(5*time.Minute), "component"); err != nil {
```

Comment:
Minor point, but this looks very similar to:

```go
				if err := utils.WaitForPodsReady(cmd.Context(), c, namespace, "envoy", time.Duration(5*time.Minute), "component"); err != nil {
					envoyFleetSpinner.Fail("Failed upgrading EnvoyFleet", err.Error())
					reportError(err)
					return err
				}
```

Perhaps in the future move this into a separate function?